### PR TITLE
Update undesirable behavior impact collection details

### PR DESCRIPTION
**Pull request by howard-e**
This addresses feedback in https://github.com/w3c/aria-at-app/issues/738#issuecomment-1965711069.

This PR does the following:
* On test run page, fixes an issue where the undesirable behavior checkboxes couldn't be tabbed between
* On test run page, replaces Impact's 'High' option with 'Severe'
* On test results table when used anywhere in the app, updates the assertion `Other behaviors that create high negative-impacts are not exhibited` to say `Other behaviors that create severe negative-impacts are not exhibited` instead
* On the test results table when used anywhere in the app, updates the undesirable behaviors section to be labelled as `Other behaviors that create negative impact:` and presents the content in a table with Behavior, Details and Impact as the columns

Note that the aria-at sourced related changes are included in an open PR on that repository: https://github.com/w3c/aria-at/pull/1016

### CODE REVIEW

File: shared/getMetrics.js
Comment (alflennik) [inline shared/getMetrics.js:269]: Since metrics are saved in the database does that mean we should have a migration to make sure the metrics are up-to-date?

File: client/components/common/TestPlanResultsTable/index.jsx
Comment (alflennik) [inline client/components/common/TestPlanResultsTable/index.jsx:92]: The hyphen here isn't grammatically correct, it should be `'Other behaviors that create severe negative impacts are not exhibited'`.

File: client/components/common/TestPlanResultsTable/index.jsx
Comment (alflennik) [inline client/components/common/TestPlanResultsTable/index.jsx:158]: For consistency, where possible can we call these unexpected behaviors?

File: shared/getMetrics.js
Comment (alflennik) [inline shared/getMetrics.js:266]: The name `highImpactFailedAssertionCount` is still present in `server/util/getMetrics.js` is there a plan to remove it from there?

**Review by alflennik — CHANGES_REQUESTED** [review]
Hey this looks great! Having been reviewing previous PRs related to assertions, these UI changes actually helped me put it all together and understand the experience we were driving towards. I'm liking how this has evolved this year.

While testing I found a situation that could lead to the "behavior cell" being blank. As I described to you offline, you have to submit the test result with an empty details field. I'll come back later and verify the fix for that.

File: shared/getMetrics.js
Comment (howard-e) [reply shared/getMetrics.js:269]: Since this hasn't yet gone to the production environment yet to affect that database, no need to write that migration. There wouldn't be any metrics that include those properties just yet.

File: client/components/common/TestPlanResultsTable/index.jsx
Comment (howard-e) [reply client/components/common/TestPlanResultsTable/index.jsx]: True, although unsure if it was intentional. This could also be [shortened in the future soon](https://github.com/w3c/aria-at/issues/1043) to: 

> Severe negative side effects do not occur

If you'd like to weigh in on that discussion.

File: shared/getMetrics.js
Comment (howard-e) [reply shared/getMetrics.js:266]: I think this comment may have been before the latest changes from main were merged in here, so this comment may be outdated. `server/util/getMetrics.js` isn't on this branch anymore.

Comment (alflennik) [reply shared/getMetrics.js:266]: You're right!

**Review by alflennik — APPROVED** [review]
Looks great, I'm definitely happy to approve! About the grammatical issue I'll leave it up to your judgement.

**Pull request merged**
